### ISSUE TITLE
Destination Host Unreachable

### ISSUE THREAD

**Issue by acrobotic** [JiapengLi/OpenWrt-HiLink-HLK-RM04#5]
Hey, Jiapeng,

Thanks for sharing your work on this cool little module.  After following your instructions, I am able to upload the image:
openwrt-ramips-rt305x-hlk-rm04-initramfs-factory.bin

But if I do a:
ping 192.168.1.1

I get:
PING 192.168.1.1 (192.168.1.1) 56(84) bytes of data.
From 192.168.16.100 icmp_seq=1 Destination Host Unreachable

Also, only the red LED is on.

If you could provide some pointers on how to debug the system that'd be great!

**Comment by JiapengLi** [comment]
> Thanks for sharing your work on this cool little module. After following your instructions, I am able to upload the image:
> openwrt-ramips-rt305x-hlk-rm04-initramfs-factory.bin
> How do you upload this bin file? This bin can be only used to install
> OpenWrt firmware from original HLK-RM04 webui.
> 
> But if I do a:
> ping 192.168.1.1
> 
> I get:
> PING 192.168.1.1 (192.168.1.1) 56(84) bytes of data.
> From 192.168.16.100 icmp_seq=1 Destination Host Unreachable
> 
> Also, only the red LED is on.
> 
> If you could provide some pointers on how to debug the system that'd be great!
> Did you replace the Uboot?
> If you can past some boot log, it is helpful.

BR.
Jiapeng

**Comment by acrobotic** [comment]
I uploaded it using HLK-RM04 webui per your instructions, and everything went okay.  But on reboot I get nothing.  I did not replace Uboot.  

I am new to OpenWrt so I don't know how to get the boot log.  Any hints how to do this?

My next plan was to replace the resistors even though I changed the mem=16M on kernel config (so it should be okay). Thanks for your patience :)

-Z

**Comment by JiapengLi** [comment]
> I am new to OpenWrt so I don't know how to get the boot log. Any hints how to do this?
> Can you log in the Serial Console?
> If so, then you can use "dmesg" command to see what's going on.
> If not, there are some troubles. The HLK-RM04 may be bricked. We need
> find other ways to fix it.
> 
> My next plan was to replace the resistors even though I changed the mem=16M on kernel config (so it should be okay). Thanks for your patience :)
> Which trunk revision did you use? Did you set the mem=16M parameter in
> kernel_menuconfig when compiled your installed frimware?
> If you don't set the the parameter, replace the resistors may fix it.

^_^

**Comment by acrobotic** [comment]
I logged in using the Serial Console (using minicom, 57600,8,n,1) and I get garbled output, which I interpret as the modules are bricked (this is okay, I appreciate the help and understand it's part of the process).

I used the same instructions you have on the README.md file.  So the trunk version is 38333.

I also followed the software method for setting the SDRAM by doing:
`make kernel_menuconfig`

and changing:
`rootfstype=squashfs,jffs2 mem=16M`

I did this _after_ compiling the firmware though, so that might have been the problem.  I did the `make` before `make kernel_menuconfig`.  Should I have done `make kernel_menuconfig` _before_ `make`?
